### DISCUSSION TITLE
Using without cstdlib

### DISCUSSION THREAD

**Discussion by amyipdev** [nothings/stb#1265] [Q&A]
Hello,

I want to use the library and had it recommended to me by a friend. However, it appears to be dependent on a stdlib implementation in stdlib.h. I have a minimal standard library (POSIX-UEFI) but not a full one, and that's not its include path.

What should I do?

**Comment by nothings** [comment]
stb libraries are not generally intended for embedded/low-memory-use applications, they're generally reliant on stdlib, so I have no idea whether the uefi stdlib supplies everything it needs. as to building it, it's the same as any library, you'll have to strip out the existing #includes and replace them with a uefi.h include, or whatever exactly is involved in that (I know nothing about it).

**Comment by amyipdev** [comment]
Is there a list of functions anywhere which need to be implemented?

**Comment by nothings** [comment]
This repository is a collection of 18 libraries, so it depends which one you want. Some of them have some notes at the #include lines about the functions needed from each header, but they're probably not 100% up to date. As said, it was written assuming a normal stdlib, so there's no need for such a list under that assumption.
